sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -637,6 +640,7 @@ def _delete_blob_options(self, delete_snapshots=False, **kwargs):
             raise ValueError("The delete_snapshots option cannot be used with a specific snapshot.")
         options = self._generic_delete_blob_options(delete_snapshots, **kwargs)
         options['snapshot'] = self.snapshot
+        options['version_id'] = kwargs.pop('version_id', None) or self.version_id
```

Comment:
If we're also supporting version override for download and delete we should add to the docstring :)

Reply:
good catch, I missed the docstr.
How do you feel about this kwargs.pop('version_id', None) or self.version_id, do you feel redundant or it's fine?

Reply:
discussed offline, remove self.version_id